aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -461,6 +461,81 @@ Tensor _cholesky_helper_cuda(const Tensor& self, bool upper) {
   }
 }
 
+template <typename scalar_t, bool upper>
+__global__
+void triu_tril_kernel(
+    scalar_t* result, scalar_t* self, int64_t k, int64_t N,
```

Comment:
cuda kernels look simpler now, don't they? :+1:

Reply:
Of course, they do. Thank you!